### PR TITLE
WFLY-10419 Use server snapshots to restore test state

### PR DESCRIPTION
**Pull request by stuartwdouglas**
https://issues.jboss.org/browse/WFLY-10419

### CODE REVIEW

**Comment by stuartwdouglas** [comment]
Retest this please

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jca/statistics/DataSourceJdbcStatisticsTestCase.java [hunk 1]
```diff
@@ -81,20 +81,6 @@ public void doSetup(final ManagementClient managementClient) throws Exception {
             }
         }
 
-        @Override
-        public void tearDown(final ManagementClient managementClient, final String containerId) throws Exception {
-            try {
-                StringBuffer sb = assertStatisticsSet(true, getStatAddr(dsAddress)).append(
```
Comment (jmesnil) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jca/statistics/DataSourceJdbcStatisticsTestCase.java]: With your change, there is no longer these assertions when the test is tear down. They should not have been there in the first place but maybe they should be moved to the end of the test instead?

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/messaging/mgmt/PooledConnectionFactoryStatisticsTestCase.java [hunk 2]
```diff
@@ -86,10 +88,6 @@ public void testStatistics() throws Exception {
 
             bean.closeConnection();
             assertEquals(0, readStatistic("InUseCount"));
-
-        } finally {
-            disableStatistics();
-            checkStatisticsAreDisabled();
```
Comment (jmesnil) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/messaging/mgmt/PooledConnectionFactoryStatisticsTestCase.java]: The call to `checkStatisticsAreDisabled();` should still be present after the try-with-resources block

**Review by jmesnil — CHANGES_REQUESTED** [review]
This looks good to me (apart the missing assertions I noticed).

Just a note that with this change, we no longer run any code corresponding to the `:remove` (and some :undefine-attribute) operations.
If there is some additional behaviour for some of these `:remove` operations, we'd add specific tests for them.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jca/statistics/DataSourceJdbcStatisticsTestCase.java (hunk 1 above)
Comment (stuartwdouglas) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jca/statistics/DataSourceJdbcStatisticsTestCase.java]: I don't think they are particularly important assertions, it is basically just a test that basic management functionality works

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/messaging/mgmt/PooledConnectionFactoryStatisticsTestCase.java (hunk 2 above)
Comment (stuartwdouglas) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/messaging/mgmt/PooledConnectionFactoryStatisticsTestCase.java]: Why? We know they are disabled by default, so when we restore they will go back to the default state.

**Comment by stuartwdouglas** [comment]
Retest this please

**Review by jmesnil — APPROVED** [review]

**Comment by stuartwdouglas** [comment]
For some reason this has exposed an Undertow bug that is causing the server to fail to shut down. I have no idea why this PR triggers it as it should be affecting everything. A core upgrade with Undertow 2.0.9.Final is needed in order to get accurate testing results.

**Comment by stuartwdouglas** [comment]
Retest this please

**Comment by stuartwdouglas** [comment]
This needs a new release of wildfly-arquillian with https://github.com/wildfly/wildfly-arquillian/commit/35be260e4f1ae24cd854a8c4b07902b50bb34fb9

**Comment by stuartwdouglas** [comment]
Retest this please

**Comment by kabir** [comment]
I have started another 4 runs of this https://ci.wildfly.org/project.html?projectId=WF_PullRequest&branch_WF_PullRequest=pull%2F11260

**Pull request merged**